include/emp/Evolve/Systematics.hpp

```diff
@@ -1810,31 +1781,645 @@ namespace emp {
+      // provided
+      if (origin_pos != -1 ){
+        double origin_time = emp::from_string<double>(row[origin_pos]);
+        tax->SetOriginationTime(origin_time);
+      }
+      if (destruction_time != "inf" && destruction_time != "unknown") {
+        tax->SetDestructionTime(emp::from_string<double>(destruction_time));
+      }
+
+      // Fill in number of current and total orgs if provided
+      if (num_orgs_pos != -1) {
+        size_t num_orgs = emp::from_string<size_t>(row[num_orgs_pos]);
+        tax->SetNumOrgs(num_orgs);
+      }
+      if (tot_orgs_pos != -1) {
+        size_t tot_orgs = emp::from_string<size_t>(row[tot_orgs_pos]);
+        tax->SetTotOrgs(tot_orgs);
+      }
+
+      // Store taxon pointer
+      taxa[id] = tax;
+      // Keep track of parents so we can link up later
+      if (emp::to_lower(ancestor_list_str) != "none") {
+        emp::vector<std::string> ancestor_split = emp::slice(ancestor_list_str, ',');
+        emp::vector<int> ancestor_list = emp::from_strings<int>(ancestor_split);
+        unlinked_parents[id] = ancestor_list[0];
+      } else {
+        // If no parent, this is a root
+        num_roots++;
+        roots.push_back(tax);
+      }
+    }
+
+    // Link up parents and offspring
+    for (auto element : unlinked_parents) {
+      taxa[element.first]->parent = taxa[element.second];
+      taxa[element.second]->offspring.insert(taxa[element.first]);
+      taxa[element.second]->num_offspring++;
+    }
+
+    // Set up depth
+    emp::vector<emp::Ptr<taxon_t>> leaves;
+    emp::vector<emp::Ptr<taxon_t>> to_explore;
+    for (auto root : roots) {
+      root->depth = 0;
+      root->total_offspring = 0;
+      for (auto offspring : root->GetOffspring()) {
+        to_explore.push_back(offspring);
+      }
+    }
+
+    // Step through all taxa and fix their
+    // bookkeeping
+    // Traversal starting at roots to ensure
+    // parent depth is correct when setting offspring
+    // depth
+    emp::Ptr<taxon_t> curr;
+    while(!to_explore.empty()) {
+      curr = to_explore.back();
+      to_explore.pop_back();
+      curr->total_offspring = 0;
+      curr->depth = curr->GetParent()->depth + 1;
+      total_depth += curr->depth;
+      for (auto offspring : curr->GetOffspring()){
+        to_explore.push_back(offspring);
+      }
+      if (curr->GetNumOff() == 0) {
+        leaves.push_back(curr);
+      }
+    }
+
+    // If we're assuming that all leave are extant,
+    // move leaves to active taxa
+    if (assume_leaves_extant) {
+      for (auto leaf : leaves) {
+        if (!Has(active_taxa, leaf)) {
+          ancestor_taxa.erase(leaf);
+          active_taxa.insert(leaf);
+        }
+      }
+    }
+
+    if (num_orgs_pos == -1) {
+      num_orgs_wrong = true;
+      for (auto tax : active_taxa) {
+        tax->SetNumOrgs(1);
+        tax->SetTotOrgs(1);
+        org_count++;
+      }
+    }
+
+    // Adjust total offspring
+    if (adjust_total_offspring) {
```

Comment:
Why not just always adjust_total_offspring? then you could not have to worry about total_offspring_wrong

Reply:
Because it's really slow and not necessary for all use cases (https://github.com/emilydolson/phylotrackpy/issues/6)